tests/functional/login.js

```diff
@@ -0,0 +1,80 @@
+const Nightmare = require('nightmare');
+const assert = require('chai').assert;
+
+const url = 'http://localhost:3333/';
+
+
+
+describe('Login tests', function() {
+
+  this.timeout('30s');
+
+  let browser = null;
+
+  beforeEach( () => {
+    browser = new Nightmare({ 
+      // to make the typed input much faster
+      typeInterval: 20,
+      // uncomment this to show the browser and the debug window
+      // openDevTools: {
+      //   mode: "detach"
+      // },
+      // show: true 
+    });
+    browser
+      .goto(url);
+  });
+
+  describe('Functional tests for login', () => {
+  
+    it('we should be redirected to the login page', done => {
+      browser
+        .end()
```

Comment:
Can you explain the purpose of the end here? I'm not sure I get it.

Reply:
See https://github.com/segmentio/nightmare#end and the api documented there. See also the `beforeEach` where we queue the url.